### PR TITLE
Prepare BDR 5.1.0 , 4.3.1 and 3.7.21

### PR DESCRIPTION
**Pull request by irionr**
Draft PR to use for future docs and release notes for PGD 5.1.0, 4.3.1 and 3.7.21

### CODE REVIEW

File: product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx [hunk 1]
```diff
@@ -6,6 +6,29 @@ navTitle: "Version 5.1.0"
...
+| PGD       | 5.1.0   | Bug fix         | Fix spurious warnings when processing sequence options (BDR-3295, RT90668) |
+| PGD       | 5.1.0   | Bug fix         | Fix upgrades for nodes with CRDTs (BDR-3302) |
+| PGD       | 5.1.0   | Feature         | Add group name in the bdr.event_summary view (BDR-3315) |
+| PGD       | 5.1.0   | Bug fix         | Adjust lag tracking parameters for LCRs from pg_stat_replication (BDR-3040) |
+| PGD       | 5.1.0   | Bug fix         | Adjust node routing options defaults based on node kind (BDR-3029)  for example witness nodes are not marked as candidates for receiving writes |
+| PGD       | 5.1.0   | Feature         | Add SLES support (BDR-3101)   SLES 12sp5 and SLES 15sp4 are now supported |
+| PGD       | 5.1.0   | Feature         | Convert all sequences to "distributed" during create node (BDR-2890)  |
+| PGD       | 5.1.0   | Feature         | Add subscription status column to the group subscription summary view. (BDR-2198)  This will help distinguish whether the NULL values is due to a node being down or subscription being disabled.|
```
Comment (djw-m) [inline product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx:20]: "whether the NULL value" or "whether NULL values" depending on how many null values we are talking about.

Comment (irionr) [reply product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx]: thanks, done

File: product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx [hunk 2]
```diff
@@ -6,6 +6,29 @@ navTitle: "Version 5.1.0"
...
+| PGD       | 5.1.0   | Bug fix         | Fix spurious warnings when processing sequence options (BDR-3295, RT90668) |
+| PGD       | 5.1.0   | Bug fix         | Fix upgrades for nodes with CRDTs (BDR-3302) |
+| PGD       | 5.1.0   | Feature         | Add group name in the bdr.event_summary view (BDR-3315) |
+| PGD       | 5.1.0   | Bug fix         | Adjust lag tracking parameters for LCRs from pg_stat_replication (BDR-3040) |
+| PGD       | 5.1.0   | Bug fix         | Adjust node routing options defaults based on node kind (BDR-3029)  for example witness nodes are not marked as candidates for receiving writes |
+| PGD       | 5.1.0   | Feature         | Add SLES support (BDR-3101)   SLES 12sp5 and SLES 15sp4 are now supported |
+| PGD       | 5.1.0   | Feature         | Convert all sequences to "distributed" during create node (BDR-2890)  |
+| PGD       | 5.1.0   | Feature         | Add subscription status column to the group subscription summary view. (BDR-2198)  This will help distinguish whether NULL values is due to a node being down or subscription being disabled.|
```
Comment (djw-m) [inline product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx:20]: NULL values are due to...

or a subscription being disabled.

(Github's not letting me suggest again, sorry)

File: product_docs/docs/pgd/3.7/bdr/release-notes.mdx [hunk 3]
```diff
@@ -5,6 +5,52 @@ originalFilePath: release-notes.md
...
+
+- Fix replication for subscriber-only node (BDE-3204, RT89814)
+
+- Fix issue where a node can be inconsistent with the group after rejoining
+  If a node had been part of a sub-group, then parted and then rejoined to the
+  group, it could be inconsistent with the group. The changes from some nodes
+  of the group would be replayed from a wrong starting point, resulting in
+  potential data-loss.
+
+- Free memory consumed by dlist node holding MQConn (BDR-3409, RT91830)
```
Comment (PJMODOS) [inline product_docs/docs/pgd/3.7/bdr/release-notes.mdx:32]: Can we maybe make this "Fix memory leak in consensus process", this is supposed to be consumable for customer, they have no idea what MQConn or dlist are :)

Comment (PJMODOS) [reply product_docs/docs/pgd/3.7/bdr/release-notes.mdx:32]: Also this one is much more important than SDW so it should be first

File: product_docs/docs/pgd/3.7/bdr/release-notes.mdx [hunk 4]
```diff
@@ -5,6 +5,52 @@ originalFilePath: release-notes.md
...
+
+### Resolved Issues
+
+- Fix join and replication when SDW and standby_slot_names are set (BDR-3176, RT89702, RT89536)
+
+- Fix upgrades for nodes with CRDTs (BDR-3302)
+
+- Fix replication for subscriber-only node (BDE-3204, RT89814)
+
+- Fix issue where a node can be inconsistent with the group after rejoining
```
Comment (PJMODOS) [inline product_docs/docs/pgd/3.7/bdr/release-notes.mdx:26]: Would put this before the SDW one

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.0_rel_notes.mdx [hunk 5]
```diff
@@ -10,6 +10,7 @@ EDB Postgres Distributed version 4.3.0 is a patch release of EDB Postgres Distri
 
 | Component | Version |  Type           |                                                   Description                                                           |
 | --------- | ------- | --------------- | ------------------------------------------------------------------------------------------------------------------------|
+| BDR       | 4.3.0   | Feature         | Add pid to the log message emitted upon writer process death (BDR-2534) |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.0_rel_notes.mdx:13]: Huh? why is this both in 4.3.0 and 4.3.1?

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.1_rel_notes.mdx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+| BDR       | 4.3.1   | Bug fix         | Adjust lag tracking parameters for LCRs from pg_stat_replication (BDR-3040) |
+| BDR       | 4.3.1   | Bug fix         | Convert all sequences to "distributed" during create node (BDR-2890) |
+| BDR       | 4.3.1   | Bug fix         | Fix streaming transactions with standby_slot_names (BDR-3248)  This could have led to a subscriber-only node get ahead of a data-node.|
+| BDR       | 4.3.1   | Bug fix         | Free memory consumed by dlist node holding MQConn (BDR-3409, RT91830)  The memory consumed by the node is merely 32 bytes but when the consensus worker handles hundreds of requests per second, sustained for hours, the memory builds up. We saw 47% of memory consumed by consensus worker when used with HARP, which executes bdr.consensus_kv_fetch() at a rate of 600 times per second. |
+| BDR       | 4.3.1   | Bug fix         | Make Raft leader step down on loss of quorum  (isolated) |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.1_rel_notes.mdx:22]: This is not a bug fix

File: product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+| Component | Version | Type            | Description                                          |
+|-----------|---------|-----------------|------------------------------------------------------|
+| PGD       | 5.1.0   | Feature         | Add pid to the log message emitted upon writer process death (BDR-2534) |
+| PGD       | 5.1.0   | Bug fix         | Fix join and replication when SDW and standby_slot_names are set (BDR-3176, RT89702, RT89536) |
+| PGD       | 5.1.0   | Bug fix         | Fix spurious warnings when processing sequence options (BDR-3295, RT90668) |
+| PGD       | 5.1.0   | Bug fix         | Fix upgrades for nodes with CRDTs (BDR-3302) |
+| PGD       | 5.1.0   | Feature         | Add group name in the bdr.event_summary view (BDR-3315) |
+| PGD       | 5.1.0   | Bug fix         | Adjust lag tracking parameters for LCRs from pg_stat_replication (BDR-3040) |
+| PGD       | 5.1.0   | Bug fix         | Adjust node routing options defaults based on node kind (BDR-3029)  for example witness nodes are not marked as candidates for receiving writes |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx:17]: Can you mention this only affects display and not behavior?

File: product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+| PGD       | 5.1.0   | Feature         | Add SLES support (BDR-3101)   SLES 12sp5 and SLES 15sp4 are now supported |
+| PGD       | 5.1.0   | Feature         | Convert all sequences to "distributed" during create node (BDR-2890)  |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx:19]: Isn't this supposed to be bugfix?

File: product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+EDB Postgres Distributed version 5.1.0 is a patch version of EDB Postgres Distributed.
+This version addresses security vulnerabilities in dependencies for PGD-Proxy and PGD-CLI.
+
+| Component | Version | Type            | Description                                          |
+|-----------|---------|-----------------|------------------------------------------------------|
+| PGD       | 5.1.0   | Feature         | Add pid to the log message emitted upon writer process death (BDR-2534) |
+| PGD       | 5.1.0   | Bug fix         | Fix join and replication when SDW and standby_slot_names are set (BDR-3176, RT89702, RT89536) |
+| PGD       | 5.1.0   | Bug fix         | Fix spurious warnings when processing sequence options (BDR-3295, RT90668) |
+| PGD       | 5.1.0   | Bug fix         | Fix upgrades for nodes with CRDTs (BDR-3302) |
+| PGD       | 5.1.0   | Feature         | Add group name in the bdr.event_summary view (BDR-3315) |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx:15]: Features should be before bugfixes

Comment (PJMODOS) [reply product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx:15]: affects other features too

File: product_docs/docs/pgd/3.7/bdr/release-notes.mdx (hunk 4 above)
Comment (PJMODOS) [reply product_docs/docs/pgd/3.7/bdr/release-notes.mdx:26]: (This is also true for 4.3.1 and 5.1.0)

File: product_docs/docs/pgd/3.7/bdr/release-notes.mdx (hunk 3 above)
Comment (PJMODOS) [reply product_docs/docs/pgd/3.7/bdr/release-notes.mdx:32]: (This is also true for 4.3.1 and 5.1.0)

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.1_rel_notes.mdx [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+| BDR       | 4.3.1   | Bug fix         | Avoid restarting sync workers.(BDR-3415)  This is to prevent the node join from failing when config changes are made that signal the restart of subscription workers. |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.1_rel_notes.mdx:23]: Not sure if I'd classify this as a bug fix, but it's not exactly feature either. EPAS sometimes uses "Enhancement" or "Change" for these kinds of things in their release notes. I've used "Enhancement" before in PGD changes too.

File: product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx [hunk 11]
```diff
@@ -0,0 +1,34 @@
+---
+title: "Release notes for EDB Postgres Distributed version 5.1.0"
+navTitle: "Version 5.1.0"
+---
+
+EDB Postgres Distributed version 5.1.0 is a patch version of EDB Postgres Distributed.
+This version addresses security vulnerabilities in dependencies for PGD-Proxy and PGD-CLI.
+
+| Component | Version | Type            | Description                                          |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx:9]: Missing from this list are:
- consistent use of tcp keepalives across all BDR connections ( 77e3df56010c2301c573ef8c63f9c043067f1b4c )
- timeout for inactive Raft connections, same reasoning and benefits as wal_sender_timeout/wal_receiver_timeout  ( 234e9f718774531be85a9f63a49c7e867bfdb8d5 )
- making backends that receive Raft messages easily identifiable ( 8d29a110ba6d35bc284afea38352cf1e5ea2206b )

File: product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx [hunk 12]
```diff
@@ -0,0 +1,34 @@
+---
+title: "Release notes for EDB Postgres Distributed version 5.1.0"
+navTitle: "Version 5.1.0"
+---
+
+EDB Postgres Distributed version 5.1.0 is a patch version of EDB Postgres Distributed.
+This version addresses security vulnerabilities in dependencies for PGD-Proxy and PGD-CLI.
```
Comment (pavanvd) [inline product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx:7]: why are we calling it a Patch version? I thought based on semver, a patch version is when the last digit is incremented. But here we are releasing 5.1.0 (or 5.1 IOW).

Comment (mwanner) [reply product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx:7]: Yes, this is a minor version, not patch.

File: product_docs/docs/pgd/5/rel_notes/index.mdx [hunk 13]
```diff
@@ -15,6 +16,7 @@ that introduced the feature.
 
 | Release Date  |   EDB Postgres Distributed   | BDR extension | PGD CLI | PGD Proxy |
 | ------------- | ---------------------------- | ------------- | ------- | --------- |
+| 2023 May 16   | [5.1.0](pgd_5.1.0_rel_notes) | 5.1.0         | 5.1.0   | 5.1.0     |
 | 2023 Mar 21   | [5.0.1](pgd_5.0.1_rel_notes) | 5.0.0         | 5.0.1   | 5.0.1     |
```
Comment (pavanvd) [inline product_docs/docs/pgd/5/rel_notes/index.mdx:20]: We have a future date for 5.0.1? Sorry, not tracking releases closely, but it seemed odd.

File: product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx (hunk 12 above)
Comment (pavanvd) [reply product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx:7]: Ok, thanks for confirming.

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.1_rel_notes.mdx [hunk 14]
```diff
@@ -0,0 +1,24 @@
+---
+title: "Release notes for EDB Postgres Distributed version 4.3.1"
+navTitle: "Version 4.3.1"
+---
+
+EDB Postgres Distributed version 4.3.1 is a patch release of EDB Postgres Distributed 4, which includes bug fixes for issues identified in previous versions.
+
+!!! Note
+    This version is required for EDB Postgres Advanced Server versions 12.14.18, 13.10.14, 14.7.0 and later.
+
```
Comment (pavanvd) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.1_rel_notes.mdx:10]: Do we have similar dependencies for Postgres Extended and/or community Postgres?

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.0_rel_notes.mdx (hunk 5 above)
Comment (irionr) [reply product_docs/docs/pgd/4/rel_notes/pgd_4.3.0_rel_notes.mdx:13]: no, this is only 4.3.1
Removed

File: product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+
+!!! Note
+    This version is required for EDB Postgres Advanced Server versions 12.15, 13.11, 14.8 and later.
+
+| Component | Version | Type            | Description                                          |
+|-----------|---------|-----------------|------------------------------------------------------|
+| PGD       | 5.1.0   | Feature         | Add pid to the log message emitted upon writer process death (BDR-2534) |
+| PGD       | 5.1.0   | Feature         | Add group name in the bdr.event_summary view (BDR-3315) |
+| PGD       | 5.1.0   | Feature         | Add SLES support (BDR-3101)   SLES 12sp5 and SLES 15sp4 are now supported |
+| PGD       | 5.1.0   | Feature         | Add Power PC Systems support |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/5/rel_notes/pgd_5.1.0_rel_notes.mdx:17]: So apparently this should not be there as we don't want to currently sell it

**Comment by irionr** [comment]
As we are releasing 3.7.21 and 4.3.1 today and 5.1.0 next week, I will abandon this in favor of :
3.7.21 - https://github.com/EnterpriseDB/docs/pull/4080
4.3.1 - https://github.com/EnterpriseDB/docs/pull/4081
5.1.0 - https://github.com/EnterpriseDB/docs/pull/4082

**Pull request closed without merging**